src/instructlab/eval/ragas.py

```diff
@@ -0,0 +1,212 @@
+    seed: int = DEFAULT_SEED
+
+
+class RagasEvaluator(Evaluator):
+    # most basic implementation, we just assume that the user will bring the existing model responses
+    name = "ragas"
+
+    def __init__(
+        self,
+        student_model: ModelConfig | None = None,
+        run_config: RunConfig | None = None,
+        openai_client: OpenAIClient | None = None,
+    ):
+        self.student_model = student_model
+        self.run_config = run_config
+        self.openai_client = openai_client
+
+    def run(
+        self,
+        dataset: List[Sample] | Path,
+        student_model: ModelConfig | None = None,
+        run_config: RunConfig | None = None,
+        openai_client: OpenAIClient | None = None,
+    ) -> EvaluationResult:
+        """
+        Evaluates the quality of model responses against a graded rubric.
+
+        When the `dataset` lacks the `response` field, then `student_model` must be provided
+        in order to generate the answers.
+
+        Args:
+            dataset (List[Sample] | Path):
+                Can be either a list of `Sample` objects or a path to a jsonl file containing
+                records matching `Sample`.
+            student_model: (StudentModelConfig):
+                When this parameter is provided, we'll attempt to use the described model in order to
+                generate the responses from the given list of questions.
+            run_config (RunConfig | None, optional):
+                Configuration to use when running evaluations. If none is provided, then
+                a default one is created containing extremely permissive settings when handling
+                timeouts. This is because by default, OpenAI tier-1 usage accounts have very high
+                rate limits resulting in heavy throttling during evaluations.
+            openai_client (openai.Client | None, optional):
+                The client to use when generating questions from the student model, must be compatible with the OpenAI API.
+                This field is required when `student_model` is provided.
+
+        Returns:
+            EvaluationResult: The results of all evaluations performed by Ragas
+        """
+        student_model = student_model if student_model else self.student_model
+        run_config = run_config if run_config else self.run_config
+        openai_client = openai_client if openai_client else self.openai_client
+
+        # ensure we are in the dataframe format
+        input_df = None
+        if isinstance(dataset, list):
+            input_df = DataFrame(dataset)
+        elif isinstance(dataset, Path):
+            input_df = read_json(dataset, orient="records", lines=True)
+        else:
+            raise TypeError(f"invalid type of dataset: {type(dataset)}")
+
+        # this should never happen, but pylint is not smart enough to detect it
+        assert input_df is not None
+
```

Comment:
@RobotSail what do you think about adding an `assert` here if `user_input` and `reference` aren't in the `input_df`, just to make sure the `.jsonl` is in the right format.

Reply:
Done